src/thunderbots/software/ai/navigator/path_planner/theta_star_path_planner.h

```diff
@@ -0,0 +1,188 @@
+     *
+     * @return true if cell is unblocked
+     */
+    bool isUnBlocked(int row, int col);
+
+    /**
+     * Returns if a cell is the destination
+     * @param row y position of cell
+     * @param col x position of cell
+     * @param dest destination cell
+     *
+     * @return true if cell is the destination
+     */
+    bool isDestination(int row, int col, CellCoordinate dest);
+
+    /**
+     * Returns heuristic value of a cell
+     * This is currently the Euclidean distance to the destination
+     * @param row y position of cell
+     * @param col x position of cell
+     * @param dest destination cell
+     *
+     * @return Euclidean distance to dest
+     */
+    double calculateHValue(int row, int col, CellCoordinate dest);
+
+    /**
+     * Traces a path from the destination back to the start
+     * and populates a vector of points with that path
+     * @param dest destination cell
+     *
+     * @return vector of points with the path from start to dest
+     */
+    std::vector<Point> tracePath(CellCoordinate dest);
+
+    /**
+     * Updates the new node's fields based on the current node, destination
+     * and the distance to the next node
+     * and checks if destination is reached
+     * @param pCurr                 current cell
+     * @param pNext                 next cell to be updated
+     * @param dest                  destination cell
+     * @param currToNextNodeDist    Euclidean distance between pCurr and pNext
+     *
+     * @return                      true if pNew is destination
+     */
+    bool updateVertex(CellCoordinate pCurr, CellCoordinate pNew, CellCoordinate dest,
+                      double currToNextNodeDist);
+
+    /**
+     * Checks for line of sight between parent cell and new cell
+     * @param curr_parent_i         parent cell's x coordinate
+     * @param curr_parent_j         parent cell's y coordinate
+     * @param new_pair              cell to check line of sight to
+     *
+     * @return                      true if line of sight from parent to new cell
+     */
+    bool lineOfSight(int curr_parent_i, int curr_parent_j, CellCoordinate new_pair);
+
+    /**
+     * Finds closest unblocked cell to currCell
+     * @param currCell  current cell
+     *
+     * @return          closest unblocked cell to currCell
+     *                  if none found, return nullopt
+     */
+    std::optional<CellCoordinate> findClosestUnblockedCell(CellCoordinate currCell);
+
+    /**
+     * Converts a cell in grid to a point on field
+     *
+     * @param row row of cell
+     * @param col col of cell
+     *
+     * @return Point on field
+     */
+    Point convertCellToPoint(int row, int col);
+
+    /**
+     * Converts a point on field to a cell in grid
+     *
+     * @param p point on field
+     *
+     * @return cell in grid
+     */
+    CellCoordinate convertPointToCell(Point p);
+
+    // only change this value
+    static constexpr int GRID_DIVISION_FACTOR = 2;
```

Comment:
Not sure if Mat mentioned this to you or posted in the project channel, but having this at `2` instead of `1` slowed down the navigator so it took up 100 + ms per tick on his machine. So we should probably change this back to `1`. Sorry about that, I should've done more comprehensive timing testing when I changed this originally.

Reply:
thanks for confirming on Mat's laptop